optaplanner-examples/src/main/java/org/optaplanner/examples/common/persistence/SolutionConverter.java

```diff
@@ -107,6 +107,10 @@ private SolutionConverter(String dataDirName,
 
     public void convertAll() {
         File[] inputFiles = inputDir.listFiles();
+        if (inputFiles == null) {
```

Comment:
same principle here: there is already a inputDir.exists() check in the constructor. It's never null, unless there a race condition in which case NPE is fine.

Reply:
This is a null check for the `inputDir.listFiles()` return value.

> Returns null if this abstract pathname does not denote a directory, or if an I/O error occurs.

Reply:
We could perhaps improve the fail-fast approach and also check if `inputDir` is a directory in the constructor.

Reply:
I like the idea of checking if it's really a directory - will change it.